utility.lua

```diff
@@ -204,10 +218,19 @@ hopper.send_item_to = function(hopper_pos, target_pos, target_node, target_inv_i
 		return send_item_to_inv(hopper_inv, target_pos, filtered_items, placer, target_inv_info, target_def)
 	end
 
-	if hopper.config.eject_button_enabled and target_def.buildable_to
-			and hopper_meta:get_string("eject") == "true" then
-		return send_item_to_air(hopper_inv, target_pos, filtered_items)
-	end
-
 	return false
 end
+
+hopper.eject_item = function(hopper_pos, target_pos)
+	if hopper.config.eject_button_enabled then
+
+		local hopper_meta = minetest.get_meta(hopper_pos)
+		if hopper_meta:get_string("eject") == "true" then
```

Comment:
```suggestion
hopper.eject_item = function(hopper_pos, target_pos)
	if not hopper.config.eject_button_enabled then
		return false
	end

	local hopper_meta = minetest.get_meta(hopper_pos)
	if hopper_meta:get_string("eject") ~= "true" then
		return false
	end
```

....
```Lua
	return send_item_to_air(hopper_inv, target_pos)
```

1. It's always helpful to have a success/failure indicator
2. Early returns make the code easier to read. Less need to keep track of indent levels.
3. The function might be more intuitive if it were named `try_eject_item` since fail/success is heavily dependent on the hopper configuration.

Reply:
All good advice. Thank you.